shithead_game_server/src/main.rs

```diff
@@ -0,0 +1,92 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use anyhow::Context;
+use futures::{SinkExt, StreamExt};
+use log::{error, warn};
+use simple_logger::SimpleLogger;
+use tokio::net::{TcpListener, TcpStream};
+use tokio_tungstenite::tungstenite::Message;
+
+const SERVER_BIND_ADDR: &str = "0.0.0.0:7522";
+
+pub struct GameServerState {}
```

Comment:
Add a comment that explains the purpose of this class, as it wasn't implemented yet.